### PR TITLE
feat: add job_pods_creation_total metric

### PR DESCRIPTION
**Pull request by dejanzele**
#### What type of PR is this?
/kind feature
/sig apps

#### What this PR does / why we need it:

Adds job_pods_creation_total metric which tracks Pods created by Job controller based on creation event (i.e. specified PodReplacementPolicy(

#### Which issue(s) this PR fixes:

Part of [kubernetes/enhancements#3939](https://github.com/kubernetes/enhancements/issues/3939)

#### Does this PR introduce a user-facing change?

```release-note
Add job_pods_creation_total metrics for tracking Pods created by the Job controller labeled by events which triggered the Pod creation
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/issues/3939
```

### CODE REVIEW

File: test/integration/job/job_test.go [hunk 1]
```diff
@@ -82,6 +82,7 @@ func TestMetricsOnSuccesses(t *testing.T) {
 		job                       *batchv1.Job
 		wantJobFinishedNumMetric  metricLabelsWithValue
 		wantJobPodsFinishedMetric metricLabelsWithValue
+		wantJobPodsCreatedMetric  metricLabelsWithValue
```
Comment (alculquicondor) [inline test/integration/job/job_test.go:85]: can you undo the changes here? I want to get rid of this test #121408

**Review by alculquicondor — COMMENTED** [review]
/approve

File: test/integration/job/job_test.go (hunk 1 above)
Comment (dejanzele) [reply test/integration/job/job_test.go]: done

**Review by alculquicondor — COMMENTED** [review]
/lgtm

**Comment by alculquicondor** [comment]
before retesting, make sure the flakes are reported in issues.
